packages/repository/src/__tests__/integration/repositories/resolve-belongs-to-metadata.integration.ts

```diff
@@ -0,0 +1,42 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/repository
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {expect} from '@loopback/testlab';
+import {
+  BelongsToDefinition,
+  Entity,
+  ModelDefinition,
+  RelationType,
+} from '../../..';
+import {resolveBelongsToMetadata} from '../../../relations/belongs-to/belongs-to.helpers';
+
+describe('resolveBelongsToMetadata', () => {
+  it('throws if the wrong metadata type is used', async () => {
+    expect(() => {
+      resolveBelongsToMetadata(Category.definition.relations[
```

Comment:
I find this style a bit too complicated. Have you considered skipping a model class/definition and using a relation definition object directly?

```ts
const metadata: unknown = {
  name: 'category',
  type: RelationType.hasOne,
  // etc.
};

expect(
  () => resolveBelongsToMetadata(metadata as BelongsToDefinition)
).to.throw(
  // ...
);
```